tests/MongoDB.Driver.Core.Tests/Core/Operations/EndTransactionOperationTests.cs

```diff
@@ -96,14 +99,19 @@ public void CommandName_should_return_expected_result()
         [Theory]
         [ParameterAttributeData]
         public void CreateCommand_should_return_expected_result(
+            [Values(true, false)] bool useRecoveryToken,
             [Values(1, 2)] int w)
         {
             var writeConcern = new WriteConcern(w);
-            var subject = CreateSubject(writeConcern: writeConcern);
+            var recoveryToken = BsonDocument.Parse("{ cake: false }");
+            var subject = useRecoveryToken ? CreateSubject(recoveryToken, writeConcern) : CreateSubject(writeConcern: writeConcern);
 
             var result = subject.CreateCommand();
 
-            result.Should().Be($"{{ fakeOperation : 1, writeConcern : {{ w : {w} }} }}");
+            var expected = useRecoveryToken
+                ? $"{{ fakeOperation : 1, writeConcern : {{ w : {w} }}, recoveryToken : {recoveryToken} }}"
+                : $"{{ fakeOperation : 1, writeConcern : {{ w : {w} }} }}";
```

Comment:
I have several problems with how expectedResult is being computed:

* use of the ternary operator results in duplicate code between the two branches
* string interpolation is inherently dangerous
* the `w` and `recoveryToken` variables are being converted to strings and then re-parsed
* the conversion to string is via an implicit call to `ToString`, which is not necessarily the inverse of `Parse` (it's somewhat coincidental that `ToString` calls `ToJson`)
* The $"" string syntax interacts somewhat unfavorablly with JSON because all the `{` and `}` characters have to be escaped

All of the above issues can be avoided using something like this:

    var expectedResult = new BsonDocument { { "fakeOperation", 1 }, { "writeConcern", w } };
    if (useRecoveryToken)
    {
        expectedResult.Add("recoveryToken", recoveryToken);
    }

Reply:
Makes sense, taking your suggestion!